stdlib/public/Concurrency/ExecutorImpl.h

```diff
@@ -118,10 +111,22 @@ static inline SwiftJobPriority swift_job_getPriority(SwiftJob *job) {
   return (SwiftJobPriority)((job->flags >> 8) & 0xff);
 }
 
-/// Allocate memory associated with a job.
+/// Allocate memory associated with a job.  Memory is allocated using a
+/// per-task allocator and will be disposed of automatically when the
+/// task is cleaned up.
```

Comment:
maybe point out here rather than just in the `swift_job_dealloc` about the stack discipline?

Because this doc does not allude to the fact that I _have to_ figure out where to dealloc so we get the right order in many cases.

Reply:
Hmmm. I suppose. I was thinking someone would be looking directly at the file, but perhaps they're using an editor like Xcode that's picking up the doc comments.